model/finance.php

```diff
@@ -191,11 +194,36 @@ static function pripis(Uzivatel $u, $castka, $poznamka = null, Uzivatel $provedl
     ]);
   }
 
+  /**
+   * Připíše aktuálnímu uživateli $u slevu ve výši $sleva
+   * @param Uzivatel $u
+   * @param float $sleva
+   * @param string|null $poznamka
+   * @param Uzivatel|null $provedl
+   */
+  public static function pripisSlevu(Uzivatel $u, $sleva, $poznamka = null, Uzivatel $provedl = null) {
```

Comment:
Jelikož finance jsou fakticky spíš "peněženka uživatele", tak nemusí být static a lze použít $this->u. (když už u toho budeš tak ani public nemusí být, v zbytku codebase se, coby default, nepoužívá)